include/comm_key.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+namespace quda
+{
+
```

Comment:
I think most, if not all, of these functions don't need to `__device__ __host__` and can instead simply be `constexpr`